### DISCUSSION TITLE
How do i prevent transport close/ping timeout disconnection reason

### DISCUSSION THREAD

**Discussion by Santosl2** [socketio/socket.io#4752] [Q&A]
I had some issues with transport close connection. After few minutes, my client is automatically disconnected from socket.io with transport close reason. 

I already increased my pong timeout to 60.000, but it doesn't work.

How i can avoid this?

**Comment by Santosl2** [comment]
i saw this solution:

https://stackoverflow.com/questions/66351714/socketio-closes-connection-with-transport-close-very-frequently-at-client-side

It's correct?

**Reply by darrachequesne** [reply]
Hmm, this should not be needed, as there is already an heartbeat mechanism.

Is the delay between disconnections random or fixed? A fixed value could indicate that something between the client and the server closes the connection.

Reference: https://socket.io/docs/v4/troubleshooting-connection-issues/#problem-the-socket-gets-disconnected

**Reply by Santosl2** [reply]
Random... If I put a high pingTimeout is it bad? Example: 2 hours

**Reply by darrachequesne** [reply]
A high pingTimeout value makes the whole heartbeat mechanism useless, so that isn't really recommended.

Reference: https://socket.io/docs/v4/server-options/#pingtimeout

Anyway, temporary disconnections are normal, especially on mobile.

Please also take a look at connection state recovery: https://socket.io/docs/v4/connection-state-recovery
